x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -358,7 +360,23 @@ public static Query doTranslate(In in, TranslatorHandler handler) {
                     set.add(handler.convert(valueOf(e), dt));
                 }
 
-                q = new TermsQuery(in.source(), fa.exactAttribute().name(), set);
+                Object value = valueOf(list.get(0));
+                if (value instanceof ZonedDateTime || value instanceof OffsetTime) {
+                    DateFormatter formatter;
+                    q = null;
+                    if (value instanceof ZonedDateTime) {
+                        formatter = DateFormatter.forPattern(DATE_FORMAT);
+                    } else {
+                        formatter = DateFormatter.forPattern(TIME_FORMAT);
```

Comment:
This is not correct, since the branch will never be accessed with an `OffsetTime` instance: as you can see in `testTranslateInExpression_WhereClause_TimeValue` in [your own commit](https://github.com/elastic/elasticsearch/pull/60377/files/43dd7e29fc555a1e5da25d1d27243ed357485bb1..f48691a17c76a0672573866bf1ac9c44c3972fb7), the field's `::time` casting is translated as a script, so the flow will not visit this branch in that case.